mxnet-bot/LabelBotFullFunctionality/LabelBot.py

```diff
@@ -0,0 +1,262 @@
+        self.all_labels = set(all_labels)
+        return set(all_labels)
+
+    def _format_labels(self, labels):
+        """
+        This method formats labels that a user specifies for a specific issue. This is meant
+        to provide functionality for the operations on labels
+        :param labels: The messy labels inputted by the user which we want to format
+        :return: Formatted labels to send for CRUD operations
+        """
+        assert self.all_labels, "Find all labels first"
+        # clean labels, remove duplicated spaces. ex: "hello  world" -> "hello world"
+        labels = [" ".join(label.split()) for label in labels]
+        labels = [label for label in labels if label.lower() in self.all_labels]
+        return labels
+
+    def add_labels(self, issue_num, labels):
+        """
+        This method is to add a list of labels to one issue.
+        It checks whether labels exist in the repo, and adds existing labels to the issue
+        :param issue_num: The specific issue number we want to label
+        :param labels: The labels which we want to add
+        """
+        labels = self._format_labels(labels)
+        issue_labels_url = 'https://api.github.com/repos/{repo}/issues/{id}/labels' \
+            .format(repo=self.repo, id=issue_num)
+
+        response = requests.post(issue_labels_url, json.dumps(labels), auth=self.auth)
+        if response.status_code == 200:
+            logging.info('Successfully added labels to {}: {}.'.format(str(issue_num), str(labels)))
+            return True
+        else:
+            logging.error('Could not add the labels to {}: {}. \nResponse: {}'
+                          .format(str(issue_num), str(labels), json.dumps(response.json())))
+            return False
+
+    def remove_labels(self, issue_num, labels):
+        """
+        This method is to remove a list of labels to one issue.
+        It checks whether labels exist in the repo, and removes existing labels to the issue
+        :param issue_num: The specific issue number we want to label
+        :param labels: The labels which we want to remove
+        :return Response denoting success or failure for logging purposes
+        """
+        labels = self._format_labels(labels)
+        issue_labels_url = 'https://api.github.com/repos/{repo}/issues/{id}/labels/' \
+            .format(repo=self.repo, id=issue_num)
+
+        for label in labels:
+            delete_label_url = issue_labels_url + label
+            response = requests.delete(delete_label_url, auth=self.auth)
+            if response.status_code == 200:
+                logging.info('Successfully removed label to {}: {}.'.format(str(issue_num), str(label)))
+            else:
+                logging.error('Could not remove the label to {}: {}. \nResponse: {}'
+                              .format(str(issue_num), str(label), json.dumps(response.json())))
+                return False
+        return True
```

Comment:
[nit] move inside `if` for consistency

Reply:
remove_label functionality as set by the requests protocol for github only allows for me to remove each label individually so I want to return True if all of the labels have been processed and removed, if any label cannot be removed then the function would quit out and return False.